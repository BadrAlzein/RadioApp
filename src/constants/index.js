//Connection
export const STATIONS_API_URL =
  "https://jobapi.teclead-ventures.de/recruiting/radios";

//API operations
export const GET = "GET"; //TO get stations

//Radio Operations
export const ADD = "ADD"; //to ADD a station
export const BACK = "BACK"; //to the last station
export const NEXT = "NEXT"; //to the NEXT station
export const CLOSE = "CLOSE"; //to CLOSE current station


//theme
export const THEME = "THEME"; //THEME request
export const DARK = "DARK"; //DARK THEME
export const LIGHT = "LIGHT"; //LIGHT THEME