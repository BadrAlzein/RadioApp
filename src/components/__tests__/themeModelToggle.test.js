import ThemeModeToggle from "./../themeModeToggle";
import configureStore from 'redux-mock-store';
import * as CONST from "./../../constants";
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { cleanup } from "@testing-library/react";
const mockStore = configureStore([]);

describe('themeModel Toggle tests', () => {
 let store;
 let component;
 let themeButton;
  beforeEach(() => {
    store = mockStore({
      theme:{
          type: CONST.DARK,
      }
    });
    component = renderer.create(
      <Provider store={store}>
        <ThemeModeToggle />
      </Provider>
    );
    themeButton= component.root.findByType('button');
    });
    afterEach(()=>{
        cleanup()
    })
    it('renderes correctly',() => {
        expect(component).toBeDefined();
        expect(themeButton).toBeDefined();
    })
    it('match snapshot',() => {
        expect(component.toJSON()).toMatchSnapshot();
    })
    it('button renders correctly with dark mode',() => {
        //the button label should be a sun icon
        expect(themeButton._fiber.stateNode.children[0].text).toBe("ðŸ”†");
    })
    it('button works correctly with clicking',() => {  
        renderer.act(() => {
            themeButton.props.onClick();
        });
       //the fliptheme payload should be in the reducers
        expect(store.getActions()[0].type).toBe(CONST.THEME);
        expect(store.getActions()[0].payload.type).toBe(CONST.LIGHT);
    })
})
