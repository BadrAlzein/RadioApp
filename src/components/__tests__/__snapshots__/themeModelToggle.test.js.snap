// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`themeModel Toggle tests button works correctly 1`] = `
<p>
  <button
    className="themeButton darkButton"
    onClick={[Function]}
  >
    ðŸ”†
  </button>
</p>
`;

exports[`themeModel Toggle tests match snapshot 1`] = `
<p>
  <button
    className="themeButton darkButton"
    onClick={[Function]}
  >
    ðŸ”†
  </button>
</p>
`;
